File: examples-per-language/php/structural/Decorator.php
<?php

use designPatternsForHumans\structural\Decorator\MilkCoffee;
use designPatternsForHumans\structural\Decorator\SimpleCoffee;
use designPatternsForHumans\structural\Decorator\VanillaCoffee;
use designPatternsForHumans\structural\Decorator\WhipCoffee;

require_once __DIR__ . '/autoload.php';

$someCoffee = new SimpleCoffee();
echo $someCoffee->getCost() . PHP_EOL;
echo $someCoffee->getDescription() . PHP_EOL;

$someCoffee = new MilkCoffee($someCoffee);
echo $someCoffee->getCost() . PHP_EOL;
echo $someCoffee->getDescription() . PHP_EOL;

$someCoffee = new WhipCoffee($someCoffee);
echo $someCoffee->getCost() . PHP_EOL;
echo $someCoffee->getDescription() . PHP_EOL;

$someCoffee = new VanillaCoffee($someCoffee);
echo $someCoffee->getCost() . PHP_EOL;
echo $someCoffee->getDescription() . PHP_EOL;

File: Python/Structural/Decorator.py
class Coffee:
    def getCost(self):
        pass

    def getDescription(self):
        pass

class SimpleCoffee(Coffee):
    def getCost(self):
        return 10

    def getDescription(self):
        return 'Simple Coffee'

class MilkCoffee(Coffee):
    _coffee = None

    def __init__(self, coffee):
        self._coffee = coffee

    def getCost(self):
        return self._coffee.getCost() + 2

    def getDescription(self):
        return self._coffee.getDescription() + ', milk'

class WhipCoffee(Coffee):
    _coffee = None

    def __init__(self, coffee):
        self._coffee = coffee

    def getCost(self):
        return self._coffee.getCost() + 5

    def getDescription(self):
        return self._coffee.getDescription() + ', whip'


class VanillaCoffee(Coffee):
    _coffee = None

    def __init__(self, coffee):
        self._coffee = coffee

    def getCost(self):
        return self._coffee.getCost() + 3

    def getDescription(self):
        return self._coffee.getDescription() + ', vanilla'

if __name__ == '__main__':
    someCoffee = SimpleCoffee()
    print someCoffee.getCost()
    print someCoffee.getDescription()

    someCoffee = MilkCoffee(someCoffee)
    print someCoffee.getCost()
    print someCoffee.getDescription()

    someCoffee = VanillaCoffee(someCoffee)
    print someCoffee.getCost()
    print someCoffee.getDescription()

    someCoffee = WhipCoffee(someCoffee)
    print someCoffee.getCost()
    print someCoffee.getDescription()